### PR TITLE
Swiften the quote removing process

### PR DESCRIPTION
**Pull request by Arkaeriit**
A more careful use of memory leads to a way
faster processing on very larges histories. I saw
a 45x speed increase on the history of my work
computer. Furthermore, the memory usage is also
reduced.

The behavior does not changes otherwise.

### CODE REVIEW

File: src/hist_file.rs [hunk 1]
```diff
@@ -56,27 +56,32 @@ fn find_unescaped(contents: &[char], target: char) -> Option<usize> {
...
-                        let mut concat: Vec<char> = vec![];
-                        concat.extend_from_slice(ret);
-                        concat.extend_from_slice(rest);
-                        _remove_quoted_strings(concat, delimiter)
+                        slices.push(ret);
+                        let rest = &contents[first_match+second_match+2..];
+                        _remove_quoted_strings(rest, delimiter, slices);
                     },
-                    None => contents
+                    None => {return;}
```
Comment (Arkaeriit) [inline src/hist_file.rs:69]: This feels a bit ugly but I don't know what is the idiomatic way to leave a function in a match statement.

Comment (nullishamy) [reply src/hist_file.rs:69]: You could use the `if-let` syntax instead.

```rust
if let Some(first_match) = find_unescaped(&contents, delimiter) {
	let ret = &contents[0..first_match];
	if let Some(second_match) = find_unescaped(&contents[first_match+1..], delimiter) {
		slices.push(ret);
        let rest = &contents[first_match+second_match+2..];
        _remove_quoted_strings(rest, delimiter, slices);	}
}

**Review by nthnd — APPROVED** [review]
LGTM

File: src/hist_file.rs (hunk 1 above)
Comment (Arkaeriit) [reply src/hist_file.rs:69]: Thank you for the tip. It is indeed way better.

**Pull request merged**